src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -279,19 +279,19 @@ public boolean hasCustomName()
      * @param itemNBT     The NBT data to match, null ignores.
      * @return The number of items removed from the inventory.
      */
-    public int clearMatchingItems(@javax.annotation.Nullable Item itemIn, int metadataIn, int removeCount, @javax.annotation.Nullable NBTTagCompound itemNBT)
+    public int clearMatchingItems(@javax.annotation.Nullable final Item itemIn, final int metadataIn, final int removeCount, @javax.annotation.Nullable final NBTTagCompound itemNBT)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "clearMatchingItems" is 26 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 45 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 181 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)